### PR TITLE
fix(draw_sw): ensure buf_h >= 1 to prevent zero-size alloc in rotated image transform

### PR DESCRIPTION
**Pull request by lion2tomato**
# fix(draw_sw): ensure buf_h >= 1 to prevent zero-size alloc in rotated image transform

## Problem

A heap-buffer-overflow (detected by KASan) occurs in `transform_argb8888` when
rendering a rotated image whose bounding box width exceeds the internal
temporary buffer size limit.

The crash was observed in a watchface renderer drawing a 352×352 ARGB8888 image
pointer (clock hand) with rotation. The KASan report shows an out-of-bounds
write at offset 1400 from a heap object of size 0.

## Root Cause

In `img_draw_core()` (`src/draw/sw/lv_draw_sw_img.c`), the temporary buffer
height is calculated as:

```c
buf_h = MAX_BUF_SIZE / buf_stride;
```

where `MAX_BUF_SIZE` is derived from the *refreshing display* resolution and
`buf_stride = blend_w * pixel_size` depends on the *blend area* width.

When the refreshing display differs from the actual render target (e.g. in
vendor-specific rendering paths where `disp_refreshing` is a small display but
the image is rendered to a larger off-screen buffer), `buf_stride` can exceed
`MAX_BUF_SIZE`. Integer division then yields `buf_h = 0`, causing `malloc(0)`
which returns a minimal (or NULL) allocation. The subsequent transform loop
writes full pixel rows into this undersized buffer, producing a heap overflow.

Concrete example from the crash:

- Image: 352×352 ARGB8888, stride = 1408
- blend_w = 352, pixel_size = 4 → buf_stride = 1408
- MAX_BUF_SIZE based on a small refreshing display = 1200
- buf_h = 1200 / 1408 = 0
- malloc(0) → buffer too small → OOB write at pixel 350 (offset 1400)

## Fix

Add a lower-bound clamp after each `buf_h` calculation to ensure at least one
row is always allocated:

```c
buf_h = MAX_BUF_SIZE / buf_stride;
buf_h = LV_CLAMP(1, buf_h, blend_h);        /* <-- added */
if(buf_h > blend_h) buf_h = blend_h;       /* <-- removed */
```

This is applied in both the `RGB565A8` branch and the general (`else`) branch
of the buffer allocation logic in `img_draw_core()`.

The fix guarantees a valid allocation even when `buf_stride > MAX_BUF_SIZE`.
The existing row-by-row loop already handles partial-height buffers correctly,
so processing one row at a time is functionally safe — just slightly slower for
the edge case.

## Changed File

- `src/draw/sw/lv_draw_sw_img.c`: two lines added (`buf_h = LV_CLAMP(1, buf_h, blend_h);`)

## Testing

- Standalone reproducer (`tests/test_transform_crash.c`) confirms the overflow
  with ASan when the fix is absent, and passes cleanly with the fix applied.
- The bug only manifests when the refreshing display resolution differs from the
  actual render target, which occurs in custom/vendor rendering paths (e.g.
  watchface renderers). In standard single-display LVGL rendering, `blend_w` is
  clipped to the display width and `MAX_BUF_SIZE` has a 4× multiplier, so
  `buf_h >= 4` always holds.

### Notes
- Update the [Documentation](https://github.com/lvgl/lvgl/tree/master/docs) if needed.
- Add [Examples](https://github.com/lvgl/lvgl/tree/master/examples) if relevant.
- Add [Tests](https://github.com/lvgl/lvgl/blob/master/tests/README.md) if applicable.
- If you added new options to `lv_conf_template.h` run [lv_conf_internal_gen.py](https://github.com/lvgl/lvgl/blob/master/scripts/lv_conf_internal_gen.py) and update [Kconfig](https://github.com/lvgl/lvgl/blob/master/Kconfig).
- Run `scripts/code-format.py` (`astyle v3.4.12` needs to installed by running `cd scripts; ./install_astyle.sh`) and follow the [Code Conventions](https://docs.lvgl.io/master/CODING_STYLE.html).
- Mark the Pull request as [Draft](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/changing-the-stage-of-a-pull-request) while you are working on the first version, and mark is as _Ready_ when it's ready for review.
- When changes were requested, [re-request review](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/requesting-a-pull-request-review) to notify the maintainers.
- Help us to review this Pull Request! Anyone can [approve or request changes](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/approving-a-pull-request-with-required-reviews).

### CODE REVIEW

File: tests/src/test_cases/test_snapshot.c [hunk 1]
```diff
@@ -523,4 +523,60 @@ void test_snapshot_extreme_size_objects(void)
...
+
+    /* Save original resolution and shrink to 10x10.
+     *   MAX_BUF_SIZE = 4 * 10 * 4(ARGB8888) = 160 bytes
+     *
+     * A 200x200 image rotated 45° has bounding box ~283px.
+     *   buf_stride = 283 * 4 = 1132 >> 160 = MAX_BUF_SIZE
+     *   buf_h = 160 / 1132 = 0  (BUG without fix) */
+    int32_t orig_hor = lv_display_get_horizontal_resolution(disp);
+    int32_t orig_ver = lv_display_get_vertical_resolution(disp);
+    lv_display_set_resolution(disp, 10, 10);
```
[the remark this answers is not in the document]
Comment (lion2tomato) [reply tests/src/test_cases/test_snapshot.c:553]: Let's set this aside for now

**Review by FASTSHIFT — DISMISSED** [review]

**Review by FASTSHIFT — APPROVED** [review]
